packages/babel-generator/src/printer.js

```diff
@@ -513,7 +513,7 @@ export default class Printer {
     );
   }
 
-  _printComment(comment) {
+  _printComment(comment, node) {
```

Comment:
As mentioned in my last comment, I think we'd be best off not allowing the printer to know anything about the code. At the moment the `Printer` class has essentially zero node-specific logic, and I think it would be better to keep it that way.

So as mentioned, we could set a flag in `startTerminatorless` for the label case, and check that flag here instead of checking the node type.